### PR TITLE
Enable async in ES|QL REST tests

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit enables async in ES|QL REST tests.

relates #102455

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java [hunk 1]
```diff
@@ -115,22 +115,26 @@ public TransportEsqlQueryAction(
...
+            request.keepAlive(),
+            request.keepOnCompletion(),
+            listener
+        );
+    }
+
     @Override
     public void execute(EsqlQueryRequest request, EsqlQueryTask task, ActionListener<EsqlQueryResponse> listener) {
-        doExecuteForked(task, request, listener);
+        ActionRunnable.wrap(listener, l -> doExecuteForked(task, request, l)).run();
```
Comment (ChrisHegarty) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java:137]: This is required to ensure parser/verifier errors are handled correctly when going async.

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks Chris!

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java [hunk 2]
```diff
@@ -46,19 +52,31 @@
...
 public class RestEsqlTestCase extends ESRestTestCase {
 
     // Test runner will run multiple suites in parallel, with some of them requiring preserving state between
     // tests (like EsqlSpecTestCase), so test data (like index name) needs not collide and cleanup must be done locally.
     private static final String TEST_INDEX_NAME = "rest-esql-test";
 
+    private static final Logger LOGGER = LogManager.getLogger(RestEsqlTestCase.class);
+
+    public static boolean shouldLog() {
+        return true; // TODO: change to false
```
[the remark this answers is not in the document]
Comment (ChrisHegarty) [reply x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java:73]: y. I'll change this before merging.

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClusterSpecIT.java [hunk 3]
```diff
@@ -50,8 +59,27 @@ public class MultiClusterSpecIT extends EsqlSpecTestCase {
     @ClassRule
     public static TestRule clusterRule = RuleChain.outerRule(remoteCluster).around(localCluster);
 
-    public MultiClusterSpecIT(String fileName, String groupName, String testName, Integer lineNumber, CsvSpecReader.CsvTestCase testCase) {
-        super(fileName, groupName, testName, lineNumber, convertToRemoteIndices(testCase));
+    @ParametersFactory(argumentFormatting = "%2$s.%3$s")
+    public static List<Object[]> readScriptSpec() throws Exception {
```
Comment (dnhatn) [inline x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClusterSpecIT.java:63]: Can we use the ParametersFactory of the super class?

**Review by dnhatn — APPROVED** [review]
LGTM. Thank you, Chris!

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClusterSpecIT.java (hunk 3 above)
Comment (ChrisHegarty) [reply x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClusterSpecIT.java:63]: I just copied this here - so as to exclude the async test scenarios.  Eventually this should be removed.

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine rerun tests

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine retest this please

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Pull request merged**